### DISCUSSION TITLE
search_dirs override when running tests

### DISCUSSION THREAD

**Discussion by NasAmin** [vectordotdev/vector#13235] [Q&A] [answered]
Hi,

We have a `transform` that uses some lua code. We are mounting this code to the vector pods using the helm chart successfully.
However when we are running tests locally that involves loading lua libraries, the tests fail to find the lua libs.

If I additionally add the local lua config directory to the `search_dirs` of the the transform then the tests pass and the lua libs are loaded successfully.

Is there a way to override/mock the `search_dirs` when running tests?

Kind regards,

Nas

**Comment by jszwedko** [comment, marked as answer]
HI @NasAmin !

One way you could do it is by providing the `search_dirs` via an environment variable like:

```toml
[transforms.lua]
type = "lua"
seach_dirs = [${SEARCH_DIRS}]
```

That you could set to `SEARCH_DIRS` to a different value when running them as tests.
